src/main/java/net/openhft/chronicle/wire/scoped/ScopedThreadLocal.java

```diff
@@ -0,0 +1,62 @@
+package net.openhft.chronicle.wire.scoped;
+
+import net.openhft.chronicle.core.Jvm;
+
+import java.lang.reflect.Array;
+import java.util.function.Supplier;
+
+public class ScopedThreadLocal<T> {
```

Comment:
Please add some javadoc - it's hard to understand what's going on. Perhaps it makes sense to ask @peter-k-lawrey and then put it into Chronicle-Core directly